JPEGImageCompression/image_compression.py

```diff
@@ -21,25 +21,271 @@
+            Square(side_length=0.1)
+            .set_color(RED)
+            .set_opacity(1)
+            .align_to(red_channel, LEFT)
+            .align_to(red_channel, UP)
+        )
+        pixel_g = (
+            Square(side_length=0.1)
+            .set_color(GREEN)
+            .set_opacity(1)
+            .align_to(green_channel, LEFT)
+            .align_to(green_channel, UP)
+        )
+        pixel_b = (
+            Square(side_length=0.1)
+            .set_color(BLUE)
+            .set_opacity(1)
+            .align_to(blue_channel, LEFT)
+            .align_to(blue_channel, UP)
+        )
+
+        self.play(FadeIn(pixel_r), FadeIn(pixel_g), FadeIn(pixel_b))
+
+        pixel_r_big = pixel_r.copy().scale(5).move_to(ORIGIN + UP * 1.5 + RIGHT * 1.7)
+        pixel_g_big = pixel_g.copy().scale(5).next_to(pixel_r_big, DOWN, buff=1)
+        pixel_b_big = pixel_b.copy().scale(5).next_to(pixel_g_big, DOWN, buff=1)
+
+        self.play(
+            TransformFromCopy(pixel_r, pixel_r_big),
+            TransformFromCopy(pixel_g, pixel_g_big),
+            TransformFromCopy(pixel_b, pixel_b_big),
+        )
+
+        eight_bits_r = (
+            Text("8 bits", font="SF Mono")
+            .scale(0.4)
+            .next_to(pixel_r_big, RIGHT, buff=0.3)
+        )
+
+        eight_bits_g = eight_bits_r.copy().next_to(pixel_g_big)
+        eight_bits_b = eight_bits_r.copy().next_to(pixel_b_big)
+
+        self.play(FadeIn(eight_bits_r), FadeIn(eight_bits_g), FadeIn(eight_bits_b))
+
+        brace = Brace(VGroup(eight_bits_r, eight_bits_g, eight_bits_b), RIGHT)
+
+        self.play(Write(brace))
+
+        twenty_four_bits = (
+            Text("24 bits / pixel", font="SF Mono").scale(0.4).next_to(brace, RIGHT)
+        )
+
+        self.play(Write(twenty_four_bits))
+
+        self.play(Transform(twenty_four_bits, twenty_four_bits.copy().shift(UP * 0.5)))
+
+        three_bytes = (
+            Text("3 bytes / pixel", font="SF Mono")
+            .scale(0.4)
+            .next_to(twenty_four_bits, DOWN, buff=0.7)
+        )
+        self.play(Write(three_bytes))
+
+        fout = []
+        for mob in self.mobjects:
+            fout.append(FadeOut(mob))
+
+        # reset scene
+        self.play(*fout)
+
+        # flower image
+        flower_image = ImageMobject("rose.jpg").scale(0.4)
+
+        dimensions = (
+            Text("2592 × 1944", font="SF Mono")
+            .scale(0.7)
+            .next_to(flower_image, DOWN, buff=0.3)
+        )
+
+        img_and_dims = Group(flower_image, dimensions).arrange(DOWN)
+        img_and_dims_sm = img_and_dims.copy().scale(0.8).to_edge(LEFT, buff=1)
+
+        self.play(FadeIn(img_and_dims))
+        self.wait()
+        self.play(Transform(img_and_dims, img_and_dims_sm))
+
+        # I don't like how this barchart looks by default, and the fields they exposed
+        # dont quite allow for what i want to do. Ask Nipun on how to approach personalization of
+        # an existing class.
```

Comment:
Probably the best way to do it right now is through inheriting the class and adding custom behavior to it as your see it, so something like 

`class CustomBarChart(BarChart)`

You may be able to tweak the look of the bar chart however, ManimCE does a fairly good job of allowing good set of customization to the existing implementations.  Before we go that route, check if any of the existing parameters can help you achieve what you want: https://docs.manim.community/en/stable/reference/manim.mobject.probability.BarChart.html?highlight=Barchart#barchart

Reply:
Yes, I checked the documentation to see what was available, and though there is great control over colors, tick distribution and such, the thing that I dislike most is the font and the axes.

The font is MathFont by default, which is absolutely not appropriate for normal text - that is, for example, in the categories names in which I thought to call them "Uncompressed" and "Compressed". Looks very weird. Will try inheriting and see what I can do. 

Last resort is just copy the BarChart code and make it my own, but wanted to avoid that if possible and see how far we could go inheriting.